source/patterns/@aws-solutions-constructs/core/lib/websocket-api-helper.ts

```diff
@@ -0,0 +1,205 @@
+  const apiGatewayRole = new iam.Role(scope, "LambdaRestApiCloudWatchRole", {
+    assumedBy: new iam.ServicePrincipal("apigateway.amazonaws.com"),
+  });
+  props.queue.grantSendMessages(apiGatewayRole);
+  const webSocketApi = buildApiGatewayV2WebSocket(scope, id, {
+    webSocketApiProps: consolidateProps(
+      buildWebSocketApiProps(
+        apiGatewayRole,
+        props.queue,
+        props.createDefaultRoute,
+        props.defaultRouteRequestTemplate
+      ),
+      props.webSocketApiProps
+    ),
+    existingWebSocketApi: props.existingWebSocketApi,
+  });
+
+  const webSocketStage = new apigwv2.WebSocketStage(scope, "Stage", {
+    stageName: "prod",
+    webSocketApi,
+    autoDeploy: true,
+  });
+
+  const apiGatewayLogGroup = buildLogGroup(scope, "LogGroup", props.logGroupProps);
+  apiGatewayLogGroup.grant(
+    apiGatewayRole,
+    ...[
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:DescribeLogGroups",
+      "logs:DescribeLogStreams",
+      "logs:PutLogEvents",
+      "logs:GetLogEvents",
+      "logs:FilterLogEvents",
+    ]
+  );
+
+  const cfnStage: apigwv2.CfnStage = webSocketStage.node.defaultChild as apigwv2.CfnStage;
+  cfnStage.addPropertyOverride("AccessLogSettings", {
+    DestinationArn: apiGatewayLogGroup.logGroupArn,
+    Format: apigateway.AccessLogFormat.clf().toString(),
+  });
+  cfnStage.addPropertyOverride("DefaultRouteSettings", {
+    DataTraceEnabled: false,
+    DetailedMetricsEnabled: true,
+    LoggingLevel: "ERROR",
+  });
+
+  addCfnSuppressRules(webSocketStage, [
+    {
+      id: "AwsSolutions-APIG1",
+      reason: "Access logging configuration has been provided as per ApiGateway v2 requirements",
+    },
+  ]);
+
+  addCfnSuppressRules(
+    apiGatewayRole.node.tryFindChild("DefaultPolicy")?.node.tryFindChild("Resource") as cdk.CfnResource,
+    [
+      {
+        id: "AwsSolutions-IAM5",
+        reason: "The APIGateway requires permissions to KMS so that it can write to an encrypted SQS queue",
+      },
+    ]
+  );
+
+  return {
+    webSocketApi,
+    webSocketStage,
+    apiGatewayRole,
+    apiGatewayLogGroup,
+  };
+}
+
+/**
+ * build ApiGateway v2 WebSocket L2 construct. If existing WebSocketApi instance is provided, it returns that instance,
+ * if not, it creates a new WebSocketApi using the user provided props.
+ *
+ * @param scope
+ * @param props
+ * @returns
+ */
+function buildApiGatewayV2WebSocket(scope: Construct, id: string, props: BuildWebSocketApiProps): apigwv2.WebSocketApi {
+  if (props.existingWebSocketApi) {
+    return props.existingWebSocketApi;
+  } else {
+    return new apigwv2.WebSocketApi(scope, `WebSocketApi${id}`, props.webSocketApiProps);
+  }
+}
+
+/**
+ * @internal This is an internal core function and should not be called directly by Solutions Constructs clients.
+ */
+export function buildWebSocketApiProps(
+  // TODO: role and sqsQueue are only used if createDefaultRoute is true, why are they required arguments?
```

Comment:
we should address this todo before merging, or remove the comment

Reply:
updated the parameters to make them optional and removed the comment.